### PR TITLE
feat: sqlalchemy_url accepts async dialect - DIA-46202

### PR DESCRIPTION
**Pull request by hadrien**
## Description 

* `sqlalchemy_url` environment variable accepts async dialect

## Related issues

* #56 
* [DIA-46202]

## Related PRs

[DIA-46202]: https://dialoguemd.atlassian.net/browse/DIA-46202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: fastapi_sqla/__init__.py [hunk 1]
```diff
@@ -51,38 +52,48 @@
 _Session = sessionmaker()
 
 
+def new_engine(*, envvar_prefix: str = None) -> Engine:
+    envvar_prefix = envvar_prefix if envvar_prefix else "sqlalchemy_"
+    lowercase_environ = {
+        k.lower(): v for k, v in os.environ.items() if k.lower() != "sqlalchemy_warn_20"
```
Comment (vicrep) [inline fastapi_sqla/__init__.py:58]: `sqlalchemy_warn_20` quésaco?

**Review by vicrep — APPROVED** [review]

File: fastapi_sqla/__init__.py (hunk 1 above)
Comment (hadrien) [reply fastapi_sqla/__init__.py:58]: Part of having fastapi-sqla sqlachemy 2.0 compliant: https://github.com/dialoguemd/fastapi-sqla/pull/48

Comment (hadrien) [reply fastapi_sqla/__init__.py:58]: -> https://github.com/dialoguemd/fastapi-sqla/blob/master/tests/conftest.py#L8

Comment (hadrien) [reply fastapi_sqla/__init__.py:58]: -> https://docs.sqlalchemy.org/en/14/changelog/migration_20.html?highlight=sqlalchemy_warn_20#migration-to-2-0-step-two-turn-on-removedin20warnings

Comment (vicrep) [reply fastapi_sqla/__init__.py:58]: Gotcha, thanks for clarifying!

**Pull request merged**